### ISSUE TITLE
Memory leak in store_prediction line in training Unet 

### ISSUE THREAD

**Issue by parisabh** [jakeret/tf_unet#118]
Hi there,

I am trying to use Unet to train a model for my data which has 3 classes, my images are of size 512 by 512 and I have 1555 images for training. The data given to the data provider is of size(1555,512,512) for both image and labels. 
I do process the labels to make the one hot coded. Hence, after reshaping the image is (1,512,512,1) and the label is (1,512,512,3) per instance (calling next function)

when I try to train the model, my simulation gets killed in python 2.7, tensorflow 1.3. First I thought there is a memory leak in the FileWriter but then I realized the issue is with the store_prediction.

I don't get any error message, it just kills the process and exits python. 

I was wondering if you have encountered such an issue before. I am trying to train my model on cpu. Thanks a lot.

**Comment by jakeret** [comment]
Hmm no I haven't seen that before. You could try to set the number of verification images to, say, 1.
After you create the trainer instance call:

`trainer.verification_batch_size = 1`

It could also be that something is not quite right with the code writing out the prediction when dealing with >2 classes. Might be something worth investigating

**Comment by parisabh** [comment]
Thanks for the quick reply. Yes, prediction works fine when I feed it only 1 verification batch. I also changed the batch size for training to 1. However, the backprop fails. Could it be because I am running it on a CPU? Also, could you please tell me which part of the backprop code could have an issue with multi-class case? Thanks a lot!

**Comment by jakeret** [comment]
How does it fail? I don't think that it depends on the hardware.

**Comment by parisabh** [comment]
Indeed it was a hardware issue, I was working with CPU RAM=4Gb, it runs fine on CPU RAM=8 Gb. I still keep the batch size=1.  Thanks a lot.
